src/wp-admin/includes/export.php

```diff
@@ -685,7 +685,7 @@ function wxr_filter_postmeta( $return_me, $meta_key ) {
 	endforeach;
 
 				$_comments = $wpdb->get_results( $wpdb->prepare( "SELECT * FROM $wpdb->comments WHERE comment_post_ID = %d AND comment_approved <> 'spam'", $post->ID ) );
-				$comments  = array_map( 'get_comment', $_comments );
+				$comments  = array_filter( array_map( 'get_comment', $_comments ) );
```

Comment:
Oh, there's one small improvement here. Instead of relying on filtered response to be falsy, it can explicitly look to see if it is a `WP_Comment` object:


```suggestion
				$comments  = array_filter(
					array_map( 'get_comment', $_comments ),
					static function ( $comment ) {
						return $comment instanceof WP_Comment;
					}
				);
```